environment-dev.yml

```diff
@@ -2,54 +2,43 @@ name: pyjanitor-dev
 channels:
   - conda-forge
 dependencies:
-  - python=3.9
+  - python>=3.6
+  # required
+  - pandas
+  - pandas-flavor
+  - multipledispatch
+  - scipy
+  # optional
   - biopython
-  - black=22.1.0
-  - bump2version=1.0.1
-  - cairo
-  - conda
-  - darglint
-  - doc8
-  - flake8
+  - natsort
+  - pyspark>=3.2.0
+  - rdkit
+  - tqdm
+  - unyt
+  - xarray
+  # io
+  - openpyxl
+  - xlrd
+
+  # testing
+  - pytest
+  - pytest-cov
+  - pytest-xdist
   - hypothesis
+  - pandas-vet
   - interrogate
-  - ipykernel
-  - ipython
+
+  # dev tools
+  - pip-compile
+  - pre-commit
+  - pylint
   - isort
```

Comment:
We could drop `pylint` and `isort` since we start to use pre-commit.
once we drop these dependencies Makefile also should be updated.

Reply:
> I feel this is triggered because in pandas 3.6, the order of a dictionary is not assured; compared to 3.7 and above, where the order of a dictionary is assured. WIth this, I suggest we make the minimum python version 3.8, as pandas minimum version is 3.8

conda will install python from a high version to start
if there is an error happening, then install a lower version instead.
And the python version is 3.10.
Does it seem pyjanitor can't work well with py3.10?
```
 + python                               3.10.5  h582c2e5_0_cpython   conda-forge/linux-64      30MB
```
https://github.com/pyjanitor-devs/pyjanitor/runs/7332617267?check_suite_focus=true#step:3:834

Reply:
Pandas supports 3.10 so we should as well

Reply:
Having `pylint` and `isort` binaries in the environment helps with those who use VSCode for code checking. Could we ensure that these stay in the environment while also annotating with comments that they are present for developer convenience?

Reply:
I already keep IDE code style checking packages (black, pylint, isort)

Reply:
I also made a PR #1231 making similar changes. I agree with keeping black/isort, for pylint: we don't enforce it via pre-commit, instead we use flake8 as our main linting tool. So I don't see the need for keeping it around.